### PR TITLE
[buildins] PlayMedia: add option to ignore watched/played items

### PR DESCRIPTION
**Pull request by InuSasha**
additional option for PlayMedia: nowatched
effects only dir(sets,folder,tvshows...) items, single items are ignored.

if only watched or played items in the list, a notification is displayed.

### CODE REVIEW

**Comment by Razzeee** [comment]
Can you give us some usecases for this?

**Comment by InuSasha** [comment]
yes.
i plan to make the previews on homescreen configurable in the confluence-skin. But theses can include movie-sets or tvshows (not the episodes). without this pr, you see always the first movie/episode, and you have to skip to the next. (Can be many in some tvshows)

File: xbmc/interfaces/builtins/PlayerBuiltins.cpp [hunk 1]
```diff
@@ -430,19 +435,27 @@ static int PlayMedia(const std::vector<std::string>& params)
       items.Sort(SortByLabel, SortOrderAscending);
 
     int playlist = containsVideo? PLAYLIST_VIDEO : PLAYLIST_MUSIC;;
-    if (containsMusic && containsVideo) //mixed content found in the folder
+    if ((containsMusic && containsVideo) || nowatched) // only needed on mixed content or unwanted watched/played content
```
Comment (Montellese) [inline xbmc/interfaces/builtins/PlayerBuiltins.cpp]: I would use two loops to simplify the whole interference between `nowatched`, `containsMusic` and `containsVideo`. So first remove all watched items and then use the existing loop as a second loop.

**Comment by phil65** [comment]
@InuSasha do you have a forum account? If yes, how is it called?

File: xbmc/interfaces/builtins/PlayerBuiltins.cpp [hunk 2]
```diff
@@ -430,19 +435,27 @@ static int PlayMedia(const std::vector<std::string>& params)
...
           items.Remove(i);
       }
     }
 
-    g_playlistPlayer.ClearPlaylist(playlist);
-    g_playlistPlayer.Add(playlist, items);
-    g_playlistPlayer.SetCurrentPlaylist(playlist);
-    g_playlistPlayer.Play(playOffset);
+    if (items.IsEmpty())
+      CGUIDialogKaiToast::QueueNotification(g_localizeStrings.Get(16003),g_localizeStrings.Get(2080));
```
Comment (Montellese) [inline xbmc/interfaces/builtins/PlayerBuiltins.cpp]: Space after the comma between the two parameters.

**Comment by InuSasha** [comment]
solved (sorry i overseen it)

@phil65 inusasha (since 5 minutes)

**Comment by Razzeee** [comment]
So your trying to do recently added tvshows and when you click on them it jumps to the first unwatched item?
If it's not that, I'm sorry but I don't understand :/
Screenshots might help.

**Comment by InuSasha** [comment]
in a later step, the user will able to select an own playlist for this three previews. Theses playlist can include the sets and tvshows.
This pr only creates a part of the technical basement for it. (for example the onclick pr is the other)

File: xbmc/interfaces/builtins/PlayerBuiltins.cpp [hunk 3]
```diff
@@ -375,6 +377,7 @@ static int PlayMedia(const std::vector<std::string>& params)
   // ask if we need to check guisettings to resume
   bool askToResume = true;
   int playOffset = 0;
+  bool nowatched = false;
```
Comment (Razzeee) [inline xbmc/interfaces/builtins/PlayerBuiltins.cpp]: Not really happy about the naming here. Also your camelCaseing is missing.

**Comment by InuSasha** [comment]
@Razzeee i am not happy about the name, too. But i had not a better idea. i was thinking about "watched", but this can confuse with the unwatched as parameter. better possibilities are welcome.

**Comment by InuSasha** [comment]
add camel-case to noWatched variable

**Comment by mkortstiege** [comment]
In `CGUIWindowVideoNav` we already have code to select the first unwatched item. It might make sense to factor that out and re-use it for this one?

**Comment by InuSasha** [comment]
Good Advice, i will rework, and make an new PR
Thanks

**Comment by mkortstiege** [comment]
Just update it in this branch and re-open the PR. You don't have to create a new one for it.

**Comment by InuSasha** [comment]
@mkortstiege the method is a private one, and not usable in this class. To use it, i have to move it to an other location. I think CFileItemList is the best place for it.

**Comment by mkortstiege** [comment]
Not sure if `CFileItemList` is the right spot. I'd somewhat expected it to sit in `CUtil` or a newly created video related utility class in case nothing similar exists right now.

**Comment by InuSasha** [comment]
@mkortstiege CFileItemList is the wrong place. i see it the last days. But the CGUIWindowVideoNav::GetFirstUnwatchedItemIndex is not usable at my scenario.
Actual, i think CUtil or CVideoUtil class, is a good choice.

**Pull request closed without merging**